---
title: "Find-cluster-insertions"
output: html_document 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Prerequisites

**Note** mapping quality ignored, i.e. we get ambiguous and unambiguous insertions, which should not be a problem with long-reads where ambiguous reads are rare. Also our estimate is conservative, i.e. we may slightly overestimate the number of cluster insertions.


## find reads supporint cluster insertions 
```{bash find_reads, eval=FALSE}
bed="../raw-pirnacluster/cluster-ovaries-ere.ms2000.mq1.th10.bin500.bed"
scr="/Users/rokofler/dev/te-tools/longread/find-lr-bedinsertion.py"

python $scr --sid "dere,R1,G20,run1" --tes ../pele-insertions/Derecta_R1_G20_r1.pele --bed $bed > Derecta_R1_G20_r1.cli
python $scr --sid "dere,R2,G18,run1" --tes ../pele-insertions/Derecta_R2_G18_r1.pele --bed $bed > Derecta_R2_G18_r1.cli
python $scr --sid "dere,R2,G21,run1" --tes ../pele-insertions/Derecta_R2_G21_r1.pele --bed $bed > Derecta_R2_G21_r1.cli
python $scr --sid "dere,R2,G21,run2" --tes ../pele-insertions/Derecta_R2_G21_r2.pele --bed $bed > Derecta_R2_G21_r2.cli
python $scr --sid "dere,R2,G21,run3" --tes ../pele-insertions/Derecta_R2_G21_r3.pele --bed $bed > Derecta_R2_G21_r3.cli
python $scr --sid "dere,R2,G26,run1" --tes ../pele-insertions/Derecta_R2_G26_r1.pele --bed $bed > Derecta_R2_G26_r1.cli
python $scr --sid "dere,R2,G51,run1" --tes ../pele-insertions/Derecta_R2_G51_r1.pele --bed $bed > Derecta_R2_G51_r1.cli
python $scr --sid "dere,R2,G51,run2" --tes ../pele-insertions/Derecta_R2_G51_r2.pele --bed $bed > Derecta_R2_G51_r2.cli
python $scr --sid "dere,R4,G25,run1" --tes ../pele-insertions/Derecta_R4_G25_r1.pele --bed $bed > Derecta_R4_G25_r1.cli
python $scr --sid "dere,R4,G51,run1" --tes ../pele-insertions/Derecta_R4_G51_r1.pele --bed $bed > Derecta_R4_G51_r1.cli
cat Derecta_R* > all.cli
```

## find cluster insertions
ie aggregate reads supporting cluster insertions 
```{bash eval=FALSE}
python ~/dev/te-tools/longread/group-cluster-insertions.py --cli all.cli --pos-tol 20 > cluster-insertions.txt
```

## Cross link cluster insertions with tileing (step 03...)
```{bash eval=FALSE}
python ~/dev/te-tools/longread/crosslink-cli-with-tileing.py --cli cluster-insertions.txt --ts 5000 --tile ../03-clustertiles/piRNA-tileing.5kb.pirnanorm.txt > cluster-insertions-tile5kbpirnanorm.txt
```

# some statistics
## number of cluster insertions for generations <30 - read support 1
```{bash eval=FALSE}
[0,6930]rokofler%cat cluster-insertions-tile5kbpirnanorm.txt |perl -pe 's/G//'|awk '$3<30'| awk '$6>0'|awk '{print $2}'|sort|uniq -c
   8 R1
  23 R2
   4 R4
```
## number of cluster insertions for generations <30 - read support 2
```{bash eval=FALSE}
[0,6924]rokofler%cat cluster-insertions-tile5kbpirnanorm.txt |perl -pe 's/G//'|awk '$3<30'| awk '$6>1'|awk '{print $2}'|sort|uniq -c
   4 R1
   3 R2
   2 R4
```
## story - with final set of early cluster insertions
A cluster insertion may be necessary but is NOT sufficient to trigger the silencing of a newly invading TE!
```{bash eval=FALSE}
[0,6935]rokofler%cat cluster-insertions-tile5kbpirnanorm.txt |perl -pe 's/G//'|awk '$3<30'| awk '$6>1'|sort -k2,2 -k3,3n
dere	R1	20	contig_232	1319408	6	2021	F	fwd	1	39.3993230378	0.615584415584	0.513006495993
dere	R1	20	contig_26	428194	2	2437	R	rev	1	809.937512162	-0.875418535599	0.0576549918816
dere	R1	20	contig_422	1690050	3	2907	FR	fwd	1	1049.45469546	0.822135543637	0.608006580899
dere	R1	20	contig_422	9830474	3	2907	FR	fwd	1	2291.91490585	-0.207045901054	0.0573226806672
dere	R2	18	contig_508	579832	7	2389	R	rev	1	9.10789545548	-0.47191011236	0.841759980834
dere	R2	21	contig_232	3074029	2	2825	R	fwd	1	225.855340115	-0.95831445401	0.0246872804034
dere	R2	21	contig_508	579833	5	1559	F	rev	1	9.10789545548	-0.47191011236	0.841759980834
dere	R4	25	contig_513	12838058	3	1434	R	fwd	1	1128.25334154	1.0	0.372189441948
dere	R4	25	contig_513	28883839	3	2613	F	rev	1	24.3047771986	-0.473684210526	0.243747914067
[0,6936]rokofler%cat cluster-insertions-tile5kbpirnanorm.txt |perl -pe 's/G//'|awk '$3<30'| awk '$6>1'|sort -k2,2 -k3,3n > final-early-cluster-insertions.txt
```

## TSD fits of contig_508
According to https://academic.oup.com/nar/article/36/19/6199/2409963?login=true
the p-element has a characteristic TSD motif.
in contig_508 we have CTTTGGAC which is in prety good agreement with the publication

## read supporting the R2 contig_508 insertion
```{bash eval=FALSE}
[0,6942]rokofler%cat all.cli|awk '$2=="R2"'|awk '$5=="contig_508"' |awk '$6> 579800' |awk '$6<579900'    
dere	R2	G18	run1	contig_508	579830	-0..1805	R	rev	1
dere	R2	G18	run1	contig_508	579840	-518..2907	F	rev	1
dere	R2	G18	run1	contig_508	579840	-1793..2907	F	rev	1
dere	R2	G18	run1	contig_508	579830	-0..486	R	rev	1
dere	R2	G18	run1	contig_508	579830	-0..481	R	rev	1
dere	R2	G18	run1	contig_508	579830	-0..1837	R	rev	1
dere	R2	G18	run1	contig_508	579830	-0..970	R	rev	1
dere	R2	G21	run1	contig_508	579830	+0..484	F	rev	1
dere	R2	G21	run1	contig_508	579834	-1955..2907	F	rev	1
dere	R2	G21	run1	contig_508	579833	-0..1559	R	rev	1
dere	R2	G21	run3	contig_508	579840	-2591..2907	F	rev	1
dere	R2	G21	run3	contig_508	579830	-0..558	R	rev	1
```

# late cluster insertions
```{bash eval=FALSE}
[0,6909]rokofler%cat cluster-insertions-tile5kbpirnanorm.txt |perl -pe 's/G//'|awk '$3>30'| awk '$6>1'|sort -k2,2 -k3,3n > final-late-cluster-insertions.txt
[0,6910]rokofler%cat final-late-cluster-insertions.txt
dere	R2	51	contig_232	1576847	4	1360	FR	fwd	1	13.0989957113	-0.796875	0.505474477127
dere	R2	51	contig_232	3062744	6	2907	FR	rev	1	41.9577206376	0.0439024390244	0.307307718058
dere	R2	51	contig_232	754551	4	2907	FR	fwd	1	50.9632801891	-0.588353413655	0.32822185452
dere	R2	51	contig_26	481979	22	1360	FR	fwd	1	25.9421516625	0.763313609467	0.926886864318
dere	R2	51	contig_26	482381	2	2907	FR	fwd	1	25.9421516625	0.763313609467	0.926886864318
dere	R2	51	contig_26	865152	3	1274	R	rev	1	4.55394772774	0.685393258427	1.22437815394
dere	R2	51	contig_26	866569	3	1321	R	fwd	1	4.55394772774	0.685393258427	1.22437815394
dere	R2	51	contig_321	79082	2	2402	F	rev	1	57.1034344288	0.833333333333	0.378366860072
dere	R2	51	contig_422	13792453	3	2798	F	rev	1	106.224668346	-0.0915221579961	0.820159378768
dere	R2	51	contig_422	15116827	2	1358	F	fwd	1	61.5038783005	-0.976705490849	0.691259255173
dere	R2	51	contig_422	1694404	7	2907	FR	fwd	1	1049.45469546	0.822135543637	0.608006580899
dere	R2	51	contig_422	4340678	3	2326	R	fwd	1	341.955423197	-0.613048032321	0.297575372817
dere	R2	51	contig_422	4346863	6	2630	R	rev	1	598.716206318	0.77454918383	0.371931939539
dere	R2	51	contig_422	4787556	9	2907	R	fwd	1	3.78642844778	-0.891891891892	1.01238700398
dere	R2	51	contig_422	4794666	12	2907	FR	fwd	1	0.716351327959	-1.0	1.45941503171
dere	R2	51	contig_422	4795049	30	2907	FR	fwd	1	0.921023135948	-0.888888888889	0.0
dere	R2	51	contig_422	4797981	11	1360	FR	rev	1	0.921023135948	-0.888888888889	0.0
dere	R2	51	contig_422	4798066	26	2907	FR	rev	1	0.921023135948	-0.888888888889	0.0
dere	R2	51	contig_422	4815797	19	1430	FR	rev	1	1.7397103679	-0.470588235294	0.200311867097
dere	R2	51	contig_422	4839907	6	1360	FR	fwd	1	na	na	na
dere	R2	51	contig_422	4845733	3	1360	FR	fwd	1	1.12569494394	-1.0	0.0
dere	R2	51	contig_422	7599794	2	2359	F	fwd	1	8620.62304861	-0.999952515774	2.90636198348
dere	R2	51	contig_422	7600379	2	1360	FR	fwd	1	5513.24449178	-0.999851505364	2.02179124395
dere	R2	51	contig_422	7601107	6	828	FR	fwd	1	5513.24449178	-0.999851505364	2.02179124395
dere	R2	51	contig_422	9832877	22	1360	FR	rev	1	2291.91490585	-0.207045901054	0.0573226806672
dere	R2	51	contig_512	298960	2	2520	R	fwd	1	488.960949284	-0.763917957304	0.388423911396
dere	R2	51	contig_512	545785	8	1360	FR	rev	1	13.5595072792	-0.418867924528	0.308404685945
dere	R2	51	contig_512	981923	3	1358	FR	fwd	1	67.3881927802	0.293849658314	0.444731890198
dere	R2	51	contig_513	13950420	10	2907	FR	fwd	1	370.097796795	0.994746301673	0.679836266209
dere	R2	51	contig_513	28507645	3	2907	FR	rev	1	75.0122176277	-0.538881309686	0.343782167541
dere	R2	51	contig_513	3131969	3	2907	FR	fwd	1	621.844120621	-0.998848021065	2.11945218187
dere	R2	51	contig_513	8762910	3	2907	FR	fwd	1	40.6785218377	-0.994968553459	0.908080464172
dere	R2	51	contig_76	131645	4	2905	FR	rev	1	14.8898740312	-0.285223367698	0.655315798887
dere	R2	51	contig_76	149902	3	2907	R	rev	1	119.477167913	-0.844111349036	0.708769441522
dere	R4	51	contig_164	73509	3	2907	FR	fwd	1	62.5272373404	0.736497545008	0.367839467893
dere	R4	51	contig_422	7599791	5	2851	F	fwd	1	8620.62304861	-0.999952515774	2.90636198348
dere	R4	51	contig_447	11690493	10	2907	FR	rev	1	6273.60025846	0.994894297273	0.0123871572518
dere	R4	51	contig_512	1388771	3	2907	FR	fwd	1	10.1824224474	0.376884422111	0.102672414291
dere	R4	51	contig_513	29119347	2	2907	FR	fwd	1	14.0200188472	-0.43795620438	0.54683677587
```
### interesting one cluster insertions contig_422 found in both replicates R2 and R4

## cluster insertion in contig_508 is gone at later generations

Note that minimum read count requirement is reduced to 1 read here!
```{bash eval=FALSE}
[0,6912]rokofler%cat cluster-insertions-tile5kbpirnanorm.txt |perl -pe 's/G//'|awk '$3>30'| awk '$6>0'|sort -k2,2 -k3,3n|grep contig_508
dere	R2	51	contig_508	1899498	1	1093	F	rev	1	904.291215645	0.794941436089	0.470148602228
dere	R2	51	contig_508	1906249	1	1175	R	rev	1	32.2869777102	0.889064976228	0.658394314356
dere	R2	51	contig_508	283883	1	1259	R	rev	1	9.82424678344	0.510416666667	0.709437862635
```

# Visualize cluster insertions

## prepare input file
```{bash eval=FALSE}
contig_232	1319408	1
contig_26	428194	1
contig_422	1690050	1
contig_422	9830474	1
contig_508	579832	1
contig_232	3074029	1
contig_513	12838058	1
contig_513	28883839	1
```

## run for ovary and embryo with different window sizes
```{bash eval=FALSE}
 7135  samtools view ../raw-smallRNAmappings/ovaries.sort.bam |python ~/dev/te-tools/ere/graph-piRNA-distribution-near-cluins.py --sam - --insert-pos cluster-insertions.txt --bw 10000 --sample-id ovary > ovary-10k.txt
 7137  samtools view ../raw-smallRNAmappings/ovaries.sort.bam |python ~/dev/te-tools/ere/graph-piRNA-distribution-near-cluins.py --sam - --insert-pos cluster-insertions.txt --bw 20000 --sample-id ovary > ovary-20k.txt &
 7139  samtools view ../raw-smallRNAmappings/embryo.sort.bam |python ~/dev/te-tools/ere/graph-piRNA-distribution-near-cluins.py --sam - --insert-pos cluster-insertions.txt --bw 20000 --sample-id embryo > embryo-20k.txt &
 7140  samtools view ../raw-smallRNAmappings/embryo.sort.bam |python ~/dev/te-tools/ere/graph-piRNA-distribution-near-cluins.py --sam - --insert-pos cluster-insertions.txt --bw 10000 --sample-id embryo > embryo-10k.txt &
 7145  samtools view ../raw-smallRNAmappings/embryo.sort.bam |python ~/dev/te-tools/ere/graph-piRNA-distribution-near-cluins.py --sam - --insert-pos cluster-insertions.txt --bw 25000 --sample-id embryo > embryo-25k.txt &
 7146  samtools view ../raw-smallRNAmappings/ovaries.sort.bam |python ~/dev/te-tools/ere/graph-piRNA-distribution-near-cluins.py --sam - --insert-pos cluster-insertions.txt --bw 25000 --sample-id ovary > ovary-25k.txt &
cat embryo-10k.txt ovary-10k.txt >all-10k.txt
cat embryo-20k.txt ovary-20k.txt > all-20k.txt
cat embryo-25k.txt ovary-25k.txt > all-25k.txt
```
# Find early cluster insertions - that are supported by short read data
```{bash eval=FALSE}
cat cluster-insertions-tile5kbpirnanorm.txt |perl -pe 's/G//'|awk '$3<30'| awk '$6>0'|sort -k2,2 -k3,3n
dere	R1	20	contig_232	1319408	6	2021	F	fwd	1	39.3993230378	0.615584415584	0.513006495993
dere	R1	20	contig_26	428194	2	2437	R	rev	1	809.937512162	-0.875418535599	0.0576549918816
dere	R1	20	contig_422	1690050	3	2907	FR	fwd	1	1049.45469546	0.822135543637	0.608006580899
dere	R1	20	contig_422	7599107	1	499	F	fwd	1	8620.62304861	-0.999952515774	2.90636198348
dere	R1	20	contig_422	9830474	3	2907	FR	fwd	1	2291.91490585	-0.207045901054	0.0573226806672
dere	R1	20	contig_512	396411	1	2400	F	rev	1	17.806447295	-0.689655172414	0.371843293519
dere	R1	20	contig_512	935256	1	366	F	fwd	1	41.9577206376	-0.770731707317	0.647837892123
dere	R1	20	contig_513	28536791	1	2907	FR	rev	1	18.5227986229	0.486187845304	0.263393504801
dere	R2	18	contig_12	38871	1	5814	FR	fwd	1	1880.83157951	-0.799390608847	3.29672636549
dere	R2	18	contig_164	412854	1	1613	R	rev	1	149.717427543	-0.644565960355	11.3913511406
dere	R2	18	contig_232	3074038	1	1892	R	fwd	1	225.855340115	-0.95831445401	0.0246872804034
dere	R2	18	contig_232	474849	1	2904	FR	rev	1	205.592831124	-0.161274265804	0.340699676391
dere	R2	18	contig_508	579832	7	2389	R	rev	1	9.10789545548	-0.47191011236	0.841759980834
dere	R2	18	contig_508	8147216	1	2749	F	rev	1	112.72299825	-0.995460735361	0.0278236183984
dere	R2	18	contig_512	1624511	1	1839	F	rev	1	6.75416966362	-0.590909090909	0.361168366432
dere	R2	21	contig_164	73519	1	1927	F	rev	1	62.5272373404	0.736497545008	0.367839467893
dere	R2	21	contig_232	3074029	2	2825	R	fwd	1	225.855340115	-0.95831445401	0.0246872804034
dere	R2	21	contig_26	20781902	1	2901	FR	fwd	1	34.691871454	-1.0	0.582617996925
dere	R2	21	contig_422	1693255	1	1163	R	rev	1	1049.45469546	0.822135543637	0.608006580899
dere	R2	21	contig_422	7601105	1	2125	R	rev	1	5513.24449178	-0.999851505364	2.02179124395
dere	R2	21	contig_508	579833	5	1559	F	rev	1	9.10789545548	-0.47191011236	0.841759980834
dere	R2	21	contig_508	8144534	1	208	F	rev	1	64.5739554203	-0.987321711569	0.0647600964941
dere	R2	21	contig_508	8145201	1	2510	R	rev	1	112.72299825	-0.995460735361	0.0278236183984
dere	R2	21	contig_512	1579742	1	1902	R	rev	1	253.946545762	-0.81503123111	0.853555382906
dere	R2	21	contig_513	29095774	1	168	F	rev	1	1.48387060792	-0.51724137931	0.469696791813
dere	R2	26	contig_11	109830	1	2907	F	fwd	1	88.469389003	-0.952573742047	0.0984760480233
dere	R2	26	contig_26	20781837	1	187	F	rev	1	34.691871454	-1.0	0.582617996925
dere	R2	26	contig_26	20781911	1	2900	FR	fwd	1	34.691871454	-1.0	0.582617996925
dere	R2	26	contig_422	1693255	1	321	F	rev	1	1049.45469546	0.822135543637	0.608006580899
dere	R2	26	contig_422	4791786	1	682	F	fwd	1	0.716351327959	-1.0	1.45941503171
dere	R2	26	contig_508	1147366	1	1011	F	rev	1	216.338101044	0.508514664144	0.48163917713
dere	R4	25	contig_508	1218386	1	555	F	fwd	1	163.379270727	0.291575321015	0.189835173766
dere	R4	25	contig_512	1298530	1	836	R	rev	1	87.0366863471	-0.346266901822	0.596578435457
dere	R4	25	contig_513	12838058	3	1434	R	fwd	1	1128.25334154	1.0	0.372189441948
dere	R4	25	contig_513	28883839	3	2613	F	rev	1	24.3047771986	-0.473684210526	0.243747914067
cat cluster-insertions-tile5kbpirnanorm.txt |perl -pe 's/G//'|awk '$3<30'| awk '$6>0'|sort -k2,2 -k3,3n > earlycluster-shortreadsupport/early-cluster.txt
```

## check if popte2 insertions from R2 are in cluster
```{bash eval=FALSE}
 python ~/dev/te-tools/coldinvasion/find-bedinsertion.py --tes ../../popte2/pairup-mincov10/GGCTAC_ere_R2G20_trimmed.txt  --bed ../../raw-pirnacluster/cluster-ovaries-ere.ms2000.mq1.th10.bin500.bed  > cluster-insertions-G20.txt
 python ~/dev/te-tools/coldinvasion/find-bedinsertion.py --tes ../../popte2/pairup-mincov10/CAGGCG_ere_R2G34_trimmed.txt --bed ../../raw-pirnacluster/cluster-ovaries-ere.ms2000.mq1.th10.bin500.bed  > cluster-insertions-G34.txt
python ~/dev/te-tools/coldinvasion/find-bedinsertion.py --tes ../../popte2/pairup-mincov10/CAGGCG_ere_R2G40_trimmed.txt --bed ../../raw-pirnacluster/cluster-ovaries-ere.ms2000.mq1.th10.bin500.bed > cluster-insertions-G40.txt
 python ~/dev/te-tools/coldinvasion/find-bedinsertion.py --tes ../../popte2/pairup-mincov10/CAGGCG_ere_R2G48_trimmed.txt --bed ../../raw-pirnacluster/cluster-ovaries-ere.ms2000.mq1.th10.bin500.bed  > cluster-insertions-G48.txt
```

Filter all cluster insertions
```{bash eval=FALSE}
[0,6978]rokofler%cat cluster-insertions-G*|awk '$10>0'
1	contig_422	6770648	+	P-element	TIR	R	-	0.078	1
1	contig_513	12838167	.	P-element	TIR	F	-	0.216	1
1	contig_321	450774	-	P-element	TIR	F	-	0.097	1
1	contig_513	28448375	-	P-element	TIR	R	-	0.061	1
1	contig_513	28707214	-	P-element	TIR	R	-	0.047	1
1	contig_513	28773862	-	P-element	TIR	F	-	0.097	1
1	contig_513	28838180	+	P-element	TIR	R	-	0.060	1
1	contig_513	28994598	+	P-element	TIR	R	-	0.059	1
1	contig_508	306351	-	P-element	TIR	F	-	0.065	1
1	contig_321	806754	-	P-element	TIR	R	-	0.061	1
1	contig_232	287066	+	P-element	TIR	F	-	0.061	1
1	contig_232	1346436	+	P-element	TIR	R	-	0.054	1
1	contig_232	1576849	+	P-element	TIR	R	-	0.125	1
1	contig_164	430483	-	P-element	TIR	F	-	0.038	1
1	contig_512	36824	+	P-element	TIR	F	-	0.067	1
1	contig_512	189433	-	P-element	TIR	R	-	0.095	1
1	contig_512	266920	-	P-element	TIR	R	-	0.051	1
1	contig_512	617706	+	P-element	TIR	F	-	0.070	1
1	contig_512	1282179	-	P-element	TIR	F	-	0.051	1
1	contig_512	1363148	-	P-element	TIR	R	-	0.059	1
1	contig_512	1386886	+	P-element	TIR	F	-	0.083	1
1	contig_512	1408180	+	P-element	TIR	F	-	0.045	1
1	contig_512	1643839	+	P-element	TIR	R	-	0.065	1
1	contig_422	4787610	-	P-element	TIR	R	-	0.064	1
1	contig_422	4828269	+	P-element	TIR	R	-	0.069	1
1	contig_422	4852221	+	P-element	TIR	R	-	0.053	1
1	contig_422	6770509	+	P-element	TIR	F	-	0.090	1
1	contig_422	7599920	-	P-element	TIR	R	-	0.045	1
1	contig_422	18619091	-	P-element	TIR	F	-	0.044	1
1	contig_76	116586	+	P-element	TIR	R	-	0.056	1
1	contig_76	129788	+	P-element	TIR	F	-	0.052	1
1	contig_76	224042	+	P-element	TIR	R	-	0.049	1
1	contig_11	1745484	+	P-element	TIR	F	-	0.053	1
1	contig_11	1762924	+	P-element	TIR	F	-	0.099	1
```

## check for overlap between popte2 and long-reads
Check if some of the insertions 
```{bash eval=FALSE}
cat cluster-insertions-G*|awk '$10>0' > poptes-cluster-all-r2.txt
python ~/dev/te-tools/longread/find-cli-supported-byPopTE2.py --pt2 poptes-cluster-all-r2.txt --cli early-cluster.txt --tol 50 
# empty -> no overlap
```