---
title: "rna-seq overview"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Overview

```{bash eval=FALSE}
pyscript="/Users/rokofler/dev/te-tools/ere/expression/mRNA-overview.py"
if="/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/map-gsnap-trans"
of="/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/05-overview"


samtools view $if/gt_ovary_R1G10.sort.bam | python $pyscript --sam - --sample-id R1-G10-ov  > $of/ovary_R1G10.txt
samtools view $if/gt_ovary_R1G35.sort.bam | python $pyscript --sam - --sample-id R1-G35-ov   > $of/ovary_R1G35.txt
samtools view $if/gt_ovary_R2G10.sort.bam | python $pyscript --sam - --sample-id R2-G10-ov   > $of/ovary_R2G10.txt
samtools view $if/gt_ovary_R2G35.sort.bam | python $pyscript --sam - --sample-id R2-G35-ov   > $of/ovary_R2G35.txt
samtools view $if/gt_ovary_R4G10.sort.bam | python $pyscript --sam - --sample-id R4-G10-ov   > $of/ovary_R4G10.txt
samtools view $if/gt_ovary_R4G35.sort.bam | python $pyscript --sam - --sample-id R4-G35-ov   > $of/ovary_R4G35.txt

samtools view $if/gt_wf_R1G5.sort.bam | python $pyscript --sam - --sample-id R1-G5-wf   > $of/wf_R1G5.txt
samtools view $if/gt_wf_R2G5.sort.bam | python $pyscript --sam - --sample-id R2-G5-wf   > $of/wf_R2G5.txt
samtools view $if/gt_wf_R4G5.sort.bam | python $pyscript --sam - --sample-id R4-G5-wf   > $of/wf_R4G5.txt

samtools view $if/gt_wf_R1G15.sort.bam | python $pyscript --sam - --sample-id R1-G15-wf   > $of/wf_R1G15.txt
samtools view $if/gt_wf_R2G15.sort.bam | python $pyscript --sam - --sample-id R2-G15-wf   > $of/wf_R2G15.txt
samtools view $if/gt_wf_R4G15.sort.bam | python $pyscript --sam - --sample-id R4-G15-wf   > $of/wf_R4G15.txt

samtools view $if/gt_wf_R1G20.sort.bam | python $pyscript --sam - --sample-id R1-G20-wf   > $of/wf_R1G20.txt
samtools view $if/gt_wf_R2G20.sort.bam | python $pyscript --sam - --sample-id R2-G20-wf   > $of/wf_R2G20.txt
samtools view $if/gt_wf_R4G20.sort.bam | python $pyscript --sam - --sample-id R4-G20-wf   > $of/wf_R4G20.txt

samtools view $if/gt_wf_R1G30.sort.bam | python $pyscript --sam - --sample-id R1-G30-wf   > $of/wf_R1G30.txt
samtools view $if/gt_wf_R2G30.sort.bam | python $pyscript --sam - --sample-id R2-G30-wf   > $of/wf_R2G30.txt
samtools view $if/gt_wf_R4G30.sort.bam | python $pyscript --sam - --sample-id R4-G30-wf   > $of/wf_R4G30.txt

samtools view $if/gt_wf_R1G40.sort.bam | python $pyscript --sam - --sample-id R1-G40-wf   > $of/wf_R1G40.txt
samtools view $if/gt_wf_R2G40.sort.bam | python $pyscript --sam - --sample-id R2-G40-wf   > $of/wf_R2G40.txt
samtools view $if/gt_wf_R4G40.sort.bam | python $pyscript --sam - --sample-id R4-G40-wf   > $of/wf_R4G40.txt

samtools view $if/gt_wf_naive1.sort.bam | python $pyscript --sam - --sample-id R1-naive-wf   > $of/wf_naive1.txt
samtools view $if/gt_wf_naive2.sort.bam | python $pyscript --sam - --sample-id R2-naive-wf   > $of/wf_naive2.txt
samtools view $if/gt_wf_naive3.sort.bam | python $pyscript --sam - --sample-id R4-naive-wf   > $of/wf_naive3.txt

cat *txt|perl -pe 's/-/\t/g'
```

# Coverage plots
```{R eval=FALSE}
library(ggplot2)
theme_set(theme_bw())

#FBgn0003483_spnE FBtr0137012_mRNA 
#FBgn0260932_cuff FBtr0140282_mRNA 
#FBgn0034246_Dcr-2 FBtr0141095_mRNA
#FBgn0033686_Hen1 FBtr0142660_mRNA
#FBgn0033686_Hen1 FBtr0140302_mRNA
#FBgn0034098_krimp FBtr0142342_mRNA
#FBgn0032515_loqs FBtr0143902_mRNA 
#FBgn0031951_r2d2 FBtr0130551_mRNA
#FBgn0283442_vas FBtr0145198_mRNA
#FBgn0261266_zuc FBtr0130342_mRNA 
#FBgn0087035_AGO2_DM33463 FBtr0135961_mRNA
#FBgn0041164_armi FBtr0135197_mRNA 
#FBgn0000146_aub FBtr0130390_mRNA 
#FBgn0086251_del FBtr0141576_mRNA
#FBgn0030373_moon FBtr0139614_mRNA 
#FBgn0004872_piwi FBtr0130391_mRNA 
#Dere_tj 0       FBtr0141238_mRNA        1       250     263S1524M377S   *       0       0  
#Dere_rhino      0       FBtr0141090_mRNA
#Dere_rpl32      0       FBtr0132023_mRNA
#NM_142478.3_qin FBtr0143034_mRNA

genes<-c("PPI251","FBtr0137012_mRNA","FBtr0140282_mRNA","FBtr0141095_mRNA","FBtr0142660_mRNA","FBtr0140302_mRNA",
         "FBtr0142342_mRNA","FBtr0143902_mRNA","FBtr0130551_mRNA","FBtr0145198_mRNA","FBtr0130342_mRNA","FBtr0135961_mRNA",
         "FBtr0135197_mRNA","FBtr0130390_mRNA","FBtr0141576_mRNA","FBtr0139614_mRNA","FBtr0130391_mRNA",
         "FBtr0141238_mRNA","FBtr0141090_mRNA","FBtr0132023_mRNA","FBtr0143034_mRNA","FBtr0141271_mRNA")
sc<-c("PPI251","spnE","cuff","Dcr2","Hen1","Hen1","krimp","loqs","r2d2","vas",
      "zuc","ago2","armi","aub","del","moon","piwi","tj","rhino","rpl32","qin","Chk2")


h<-read.table("/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/01-visualize-Psease/raw/newallforr.forr")        
# h<-read.table("/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/01-visualize-Psease/raw/all-forr.forr")

names(h)<-c("rep","time","tissue","strand","gene","pos","cov")

for(i in 1:length(genes))
{
  target<-genes[i]
  short<-sc[i]
  
  a<-subset(h,gene==target)
  a$time <- factor(a$time, levels=c("naive", "G5", "G10", "G15", "G20", "G30","G35","G40"))
  
  s<-subset(a,strand=="se")
  as<-subset(a,strand=="ase")
  
  
  plot <- ggplot() +
    geom_polygon(data=s,mapping=aes(x=pos, y=cov), fill='grey', color='grey') +
    geom_polygon(data=as, aes(x=pos, y=-cov), fill='lightgrey', color='lightgrey')+
    facet_grid(time~rep)
  outfolder<-"/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/01-visualize-Psease/graphs/"
  outfile<-paste0(outfolder,target,"_",short,".pdf")
  
  pdf(file=outfile,width=8,height=12)
  plot(plot)
  dev.off()
}
```


# Chk2 - Wilcoxon Test

```{R}
library(ggplot2)
library(tidyverse)
theme_set(theme_bw())

h<-read.table("/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/01-visualize-Psease/raw/newallforr.forr")        
# h<-read.table("/Users/rokofler/analysis/2021-Dere-Pele/analysis/2022-01-mRNA/01-visualize-Psease/raw/all-forr.forr")

names(h)<-c("rep","time","tissue","strand","gene","pos","cov")
a<-subset(h,gene=="FBtr0141271_mRNA" & strand=="se")
mc<-a %>% group_by(rep,time) %>% summarize(mcov=mean(cov))
wilcox.test(subset(mc,rep=="R2")$mcov,subset(mc,rep!="R2")$mcov)
```
